src/app/service/jobs.service.ts

```diff
@@ -9,29 +9,35 @@ import Semaphore from 'semaphore-async-await';
 
 const RETRIES = 3;
 
-export interface JobItem<T> {
-    type: string;
-    name: string;
-    icon: string;
+export abstract class JobItem<T> {
+    readonly type: string;
+    readonly name: string;
+    readonly icon: string;
+    
     currentActivity: string;
-    isResolved: boolean;
-    toast: boolean;
-    start(): Promise<T>;
-    print(): string;
+    isResolved: boolean = false;
+    toast: boolean = false;
+    toastService: ToastService;
+
+    abstract start(): Promise<T>;
+    abstract print(): string;
+
+    injectToastService(toastService: ToastService): void {
```

Comment:
Why not injecting ToastService in constructor?

Reply:
jobs are created outside of the JobService (e.g. in shareservice) and are later on pushed to jobservice for further execution.

Reply:
ok